cirq/experiments/cross_entropy_benchmarking.py

```diff
@@ -0,0 +1,347 @@
+    all_probs = []  # type: List[np.ndarray]
+    num_states = 2 ** len(qubits)
+    for circuit in circuits:
+        trial_circuit = circuit.copy()
+        trial_circuit.append(ops.measure(*qubits, key='z'))
+        res = sampler.run(trial_circuit, repetitions=repetitions)
+        res_hist = dict(res.histogram(key='z'))
+        probs = np.zeros(num_states, dtype=float)
+        for k, v in res_hist.items():
+            probs[k] = float(v) / float(repetitions)
+        all_probs.append(probs)
+    return all_probs
+
+
+def _xeb_fidelities(ideal_probs: Dict[int, np.ndarray],
+                    actual_probs: Dict[int, np.ndarray]) -> List[float]:
+    num_cycles = sorted(list(ideal_probs.keys()))
+    xeb_fidelity = []  # type: List[float]
+    for n in num_cycles:
+        xeb_fidelity.append(_compute_fidelity(ideal_probs[n], actual_probs[n]))
+    return xeb_fidelity
+
+
+def _compute_fidelity(probs_exp: np.ndarray, probs_meas: np.ndarray) -> float:
+    _, num_trials = probs_exp.shape
+    probs_min = np.zeros_like(probs_exp) + 1e-22
+
+    probs_exp = np.asarray(np.maximum(probs_exp, probs_min) / num_trials)
+    probs_meas = np.asarray(probs_meas / num_trials)
+    prob_uni = 1.0 / probs_exp.size
+
+    s_incoherent = -np.sum(prob_uni * np.log(probs_exp))
+    s_expected = -np.sum(probs_exp * np.log(probs_exp))
+    s_meas = -np.sum(probs_meas * np.log(probs_exp))
+    return float((s_incoherent - s_meas) / (s_incoherent - s_expected))
+
+
+def _random_half_rotations(qubits: Sequence[devices.GridQubit], num_layers: int
+                           ) -> List[List[ops.OP_TREE]]:
+    rot_ops = [ops.X ** 0.5, ops.Y ** 0.5,
+               ops.PhasedXPowGate(phase_exponent=0.25, exponent=0.5)]
+    num_qubits = len(qubits)
+    rand_nums = np.random.choice(3, (num_qubits, num_layers))
+    single_q_layers = []  # type: List[List[ops.OP_TREE]]
+    for i in range(num_layers):
+        single_q_layers.append(
+            [rot_ops[rand_nums[j, i]](qubits[j]) for j in range(num_qubits)])
+    return single_q_layers
+
+
+def _random_tetrahedral_rotations(qubits: Sequence[devices.GridQubit],
+                                  num_layers: int) -> List[List[ops.OP_TREE]]:
+    rot_ops = [[ops.X ** 0], [ops.X], [ops.Y], [ops.Y, ops.X]]
+    for (s_0, s_1) in [(1, 1), (1, -1), (-1, 1), (-1, -1)]:
+        rot_ops.append([ops.X ** (s_0 * 0.5), ops.Y ** (s_1 * 0.5)])
+        rot_ops.append([ops.Y ** (s_0 * 0.5), ops.X ** (s_1 * 0.5)])
+
+    num_qubits = len(qubits)
+    rand_nums = np.random.choice(12, (num_qubits, num_layers))
+    single_q_layers = []  # type: List[List[ops.OP_TREE]]
+    for i in range(num_layers):
+        rots_i = []  # type: List[ops.OP_TREE]
+        for j in range(num_qubits):
+            rots_i.extend([rot(qubits[j]) for rot in rot_ops[rand_nums[j, i]]])
+        single_q_layers.append(rots_i)
+    return single_q_layers
+
+
+def _interaction_sequence(qubits: Sequence[devices.GridQubit]
```

Comment:
Rename to `_default_interaction_sequence`

Reply:
Done.